src/planner/ompl/Planner.cpp

```diff
@@ -372,6 +374,81 @@ trajectory::InterpolatedPtr planCRRTConnect(
+                                                                    double _maxDistanceBtwValidityChecks,
+                                                                    double _timeout, size_t _maxEmptySteps,
+                                                                    trajectory::InterpolatedPtr _originalTraj)
+{
+
+  // Step 1: Generate the space information of OMPL type
+  auto si = getSpaceInformation(
+        _stateSpace, _interpolator, std::move(_dmetric), std::move(_sampler),
+        std::move(_validityConstraint), std::move(_boundsConstraint),
+        std::move(_boundsProjector), _maxDistanceBtwValidityChecks);
+
+  // Get the state space
+  auto sspace = ompl_static_pointer_cast<GeometricStateSpace>(
+        si->getStateSpace());
+
+  // Step 2: Convert the AIKIDO trajectory to Geometric Path 
+  ::ompl::geometric::PathGeometric *path = new ::ompl::geometric::PathGeometric(si);
+
+  for (size_t idx = 0; idx < _originalTraj->getNumWaypoints(); ++idx) 
+  {
+        auto ompl_state = sspace->allocState(_originalTraj->getWaypoint(idx));
+        path->append(ompl_state);
+        sspace->freeState(ompl_state);
+  }
+
+  // Step 3: Use the OMPL methods to simplify the path
+  ::ompl::geometric::PathSimplifier simplifier = ::ompl::geometric::PathSimplifier(si);
+
+  // Flag for user to know if shorten was successful
+  bool shorten_success = false;
+
+  // Set the parameters for termination of simplification process
+  ::ompl::time::point const time_before = ::ompl::time::now();
+  ::ompl::time::point time_current;
+  ::ompl::time::duration const time_limit = ::ompl::time::seconds(_timeout);
+  double empty_steps = 0;
+  
+  do 
+  {
+
+    bool const shortened = simplifier.shortcutPath(*path, 1, _maxEmptySteps, 1.0, 0.0);
+    empty_steps = (empty_steps + !shortened)*!shortened; // Increment only if shorten fails consecutively
+    time_current = ::ompl::time::now();
+    shorten_success = shorten_success || shortened; // TODO: There should be a better way to do this.
+
+  } while(time_current - time_before <= time_limit && empty_steps <= _maxEmptySteps);
+
+  // Step 4: Convert the simplified geomteric path to AIKIDO untimed trajectory
+  auto returnTraj = dart::common::make_unique<trajectory::Interpolated>(
+        std::move(_stateSpace), std::move(_interpolator));
+
+  for (size_t idx = 0; idx < path->getStateCount(); ++idx) 
+  {
+        const auto *st =
+            static_cast<GeometricStateSpace::StateType *>(
+                path->getState(idx));
+        // Arbitrary timing
+        returnTraj->addWaypoint(idx, st->mState);
+  }
+
+  std::pair <std::unique_ptr<trajectory::Interpolated>, bool> returnPair;
+  returnPair = std::make_pair(std::move(returnTraj),shorten_success); 
+  return returnPair;
+
 }
-}
-}
+//=============================================================================
+} // ns aikido
+} // ns planner
+} // ns ompl
```

Comment:
Add a unit tests for `simplifyOMPL`. The obvious tests are: (1) each error condition, (2) shortcut a two-waypoint trajectory for a large number of iterations and verify that it doesn't change, and (3) shortcut a three waypoint trajectory for a large number of iterations and verify that the trajectory gets shorter.

It would also be nice to add tests for the `GeometricPath` conversion functions, but I feel too guilty to ask about that since I am the one who asked you to implement them. 😬